File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxDiff = 0, min = prices[0];
        for(int i=1; i<prices.length; i++){
            if(prices[i] < min){
                min = prices[i];
            } else {
                maxDiff = Math.max(prices[i]-min, maxDiff);
            }
        }
        return maxDiff;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = prices[0]
        profit = 0
        for p in prices:
            if p < min_price:
                min_price = p
            elif p > min_price :
                profit = max(profit, p - min_price)
                print(profit)
        return profit
        